tests/backup/backup_upgrade_test.go

```diff
@@ -202,3 +204,501 @@ var _ = Describe("{StorkUpgradeWithBackup}", func() {
+					restoreNames = SafeAppend(&mutex, restoreNames, restoreName).([]string)
+				}
+			}
+			_ = TaskHandler(singleNSNamespaces, restoreSingleNSBackupInVariousWaysTask, Parallel)
+		})
+		// By the time the next singleNS schedule backups are taken, the px-backup upgrade would have been completed
+		Step("Restore next single namespace schedule backups", func() {
+			ctx, err := backup.GetAdminCtxFromSecret()
+			dash.VerifyFatal(err, nil, "Fetching px-central-admin ctx")
+			restoreSingleNSBackupInVariousWaysTask := func(index int, namespace string) {
+				nextScheduleBackupName, err := GetNextScheduleBackupName(singleNSScheduleNames[index], time.Duration(intervalInMins), ctx)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Fetching next schedule backup name of schedule named [%s]", singleNSScheduleNames[index]))
+				log.InfoD("Next schedule backup name [%s]", nextScheduleBackupName)
+				restoreConfigs := []struct {
+					namePrefix          string
+					namespaceMapping    map[string]string
+					storageClassMapping map[string]string
+					replacePolicy       ReplacePolicy_Type
+				}{
+					{
+						"test-restore-single-ns",
+						make(map[string]string, 0),
+						make(map[string]string, 0),
+						ReplacePolicy_Retain,
+					},
+					{
+						"test-custom-restore-single-ns",
+						map[string]string{namespace: "custom" + namespace},
+						make(map[string]string, 0),
+						ReplacePolicy_Retain,
+					},
+					{
+						"test-replace-restore-single-ns",
+						make(map[string]string, 0),
+						make(map[string]string, 0),
+						ReplacePolicy_Delete,
+					},
+				}
+				for _, config := range restoreConfigs {
+					restoreName := fmt.Sprintf("%s-%s", config.namePrefix, RandomString(4))
+					log.InfoD("Restoring next single namespace schedule backup [%s] in cluster [%s] with restore [%s] and namespace mapping %v", nextScheduleBackupName, destinationClusterName, restoreName, config.namespaceMapping)
+					if config.replacePolicy == ReplacePolicy_Retain {
+						err = CreateRestore(restoreName, nextScheduleBackupName, config.namespaceMapping, destinationClusterName, orgID, ctx, config.storageClassMapping)
+					} else if config.replacePolicy == ReplacePolicy_Delete {
+						err = CreateRestoreWithReplacePolicy(restoreName, nextScheduleBackupName, config.namespaceMapping, destinationClusterName, orgID, ctx, config.storageClassMapping, config.replacePolicy)
+					}
+					dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying restoration [%s] of next single namespace schedule backup [%s] in cluster [%s]", restoreName, nextScheduleBackupName, restoreName))
+					restoreNames = SafeAppend(&mutex, restoreNames, restoreName).([]string)
+				}
+			}
+			_ = TaskHandler(singleNSNamespaces, restoreSingleNSBackupInVariousWaysTask, Parallel)
+		})
+		// First allNS schedule backup is taken before px-backup upgrade
```

Comment:
Same

Reply:
Changed all "allNS" in comments as well to "all namespaces".